app.js

```diff
@@ -28,13 +30,18 @@ const express = require('express'),
 // initialize application.
 var app = express()
 
+if (process.env.NODE_ENV !== 'test') {
+  // register to Azure Application Insights service for telemetry purposes
+  // instrumention key is provisioned from Azure App Service application setting (env variable)
+  azureApplicationInsights.setup(process.env.APPINSIGHTS_INSTRUMENTATIONKEY).start()
```

Comment:
I think we should only add this if the `process.env.APPINSIGHTS_INSTRUMENTATIONKEY` env var exists.
That way, it will still build on Heroku and locally without having this in the `.env`.

Also, I don't want to be too prescriptive, but I think it's better to run the Azure Analtyics stuff only in 'prod' mode.

So the code would be:

```
if (process.env.NODE_ENV === 'production' && process.env.APPINSIGHTS_INSTRUMENTATIONKEY) {
// init app insights
}
```

The morgan logger we still want to run in dev mode, so we can leave that how it was before.

Reply:
sounds good!

Reply:
it's all good. Done.